async_service/asyncio.py

```diff
@@ -246,6 +255,31 @@ def cancel(self) -> None:
                 self.cancel()
                 raise DaemonTaskExit(f"Daemon task {name} exited")
 
+            any_child_done = asyncio.Event()
+            while True:
+                children_still_running = tuple(
```

Comment:
Would it make sense to move this out of the loop and then use it as the looping condition?

```
while children_still_running:
    ...
```

Reply:
I think you are right.  Since `async_fn` has completed I think it becomes impossible for new child tasks to be scheduled at this point, allowing complete removal of the loop.